azurerm/resource_arm_app_service.go

```diff
@@ -794,7 +882,7 @@ func validateAppServiceName(v interface{}, k string) (ws []string, es []error) {
 	return
 }
 
-func flattenAppServiceSiteCredential(input *web.UserProperties) []interface{} {
+func flattenAppServiceSiteCredential(input *web.User) []interface{} {
```

Comment:
we should be pulling from the `UserProperties ` object since the non-Properties object isn't actually valid in Azure (e.g. it doesn't actually match the schema and can be nil in older configurations)

Reply:
User is using composition with UserProperties, which means that UserProperties object is sub-object of User object.